pkg/block/azure/client_cache.go

```diff
@@ -0,0 +1,91 @@
+package azure
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/service"
+	"github.com/puzpuzpuz/xsync"
+	"github.com/treeverse/lakefs/pkg/block/params"
+)
+
+type ClientContainerCache struct {
+	containerToClient *xsync.MapOf[string, *container.Client]
+	params            params.Azure
+}
+
+var ErrUnknownAzureURL = errors.New("unable to parse storage account")
+
+func NewCache(p params.Azure) *ClientContainerCache {
+	return &ClientContainerCache{
+		containerToClient: xsync.NewMapOf[*container.Client](),
```

Comment:
Is there a cache invalidation / expiry policy?
Theoretically if the cache is never evacuated this could explode over time (or become less efficient?)